### PR TITLE
Add a client-only web frontend with the help of WASM

### PR DESCRIPTION
**Pull request by ErichDonGubler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,9 +5,11 @@ At this point all it can do is convert `UNSMRY` files into the MessagePack forma
 ## Building
 
 - Install Rust and Cargo. Refer to https://rustup.rs/ for rust installation;
+- Default to using a `nightly` toolchain via `rustup default nightly`.
```
Comment (yan-zaretskiy) [inline README.md]: I prefer `rustup override set nightly` per crate.

File: eclair-web/crate/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "eclair-wasm"
+version = "0.1.0"
+authors = ["Yan <mindv0rtex@users.noreply.github.com>"]
```
Comment (yan-zaretskiy) [inline eclair-web/crate/Cargo.toml]: You should probably add your name here :-)

File: eclair-web/crate/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+edition = "2018"
+
+[lib]
+crate-type = ["cdylib", "rlib"]
```
Comment (yan-zaretskiy) [inline eclair-web/crate/Cargo.toml:8]: Curious why we need both types here.

File: eclair-web/crate/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    std::io::Cursor,
+    wasm_bindgen::{prelude::wasm_bindgen, JsValue},
+};
+
+#[wasm_bindgen]
+pub fn parse_ecl_summary(smspec: &[u8], unsmry: &[u8]) -> Result<JsValue, JsValue> {
+    let summary = Summary::new(Cursor::new(smspec), Cursor::new(unsmry))
+        .map_err(|e| e.to_string())
+        .map_err(JsValue::from)?;
+    Ok(JsValue::from_serde(&summary).unwrap())
```
Comment (yan-zaretskiy) [inline eclair-web/crate/src/lib.rs:12]: I'm so glad we use `serde` again!

File: README.md (hunk 1 above)
Comment (ErichDonGubler) [reply README.md]: I was trying to remember how to do this. I actually prefer that too. :)

File: eclair-web/crate/src/lib.rs (hunk 4 above)
Comment (ErichDonGubler) [reply eclair-web/crate/src/lib.rs:12]: Serde is the freaking bomb. :)

File: README.md (hunk 1 above)
Comment (ErichDonGubler) [reply README.md]: Added a commit for this specifically.

File: eclair-web/crate/Cargo.toml (hunk 3 above)
Comment (ErichDonGubler) [reply eclair-web/crate/Cargo.toml:8]: `"cdylib"` is necessary for making WASM libs, `"rlib"` is helpful for testing -- TIL, see https://rustwasm.github.io/wasm-pack/book/tutorials/npm-browser-packages/template-deep-dive/cargo-toml.html#1-crate-type.

**Comment by ErichDonGubler** [comment]
@mindv0rtex: No longer WIP -- whatcha think? :)

**Review by yan-zaretskiy — APPROVED** [review]
LGTM

**Pull request merged**